python/spacetime_local/frame.py

```diff
@@ -161,30 +184,48 @@ def run(self):
         self.thread.start()
         self.thread.join()
 
+    def __clear(self):
+        self.__disconnected = False
+        self.__app.done = False
```

Comment:
This clear is for restarting the simulation after it's been stopped. The disconnected is just a flag to skip pulling/pushing, which we one last time before the simulation ends.